src/components/SeatContest.tsx

```diff
@@ -275,39 +292,58 @@ class SeatContest extends React.Component<Props, State> {
                 </Choice>
               )
             })}
-            <Choice
-              htmlFor="writeInCandidate"
-              isSelected={writeInCandidateIsChecked}
-              onClick={this.handleWriteInCandidateDisabledClick}
-            >
-              <ChoiceInput
-                autoFocus={writeInCandidateIsChecked}
-                id="writeInCandidate"
-                name={contest.id}
-                value="writeInCandidate"
-                onChange={this.handleSelectWriteInCandidate}
-                checked={writeInCandidateIsChecked}
-                disabled={!!vote && !writeInCandidateIsChecked}
-                className="visually-hidden"
-              />
-              <Prose>
-                {!!writeInCandidateName ? (
-                  <strong>{writeInCandidateName}</strong>
-                ) : (
-                  <em>add a write-in candidate</em>
-                )}
-              </Prose>
-            </Choice>
+            {vote
+              .filter(c => c.isWriteIn)
+              .map(candidate => {
+                return (
+                  <Choice key={candidate.id} htmlFor={candidate.id} isSelected>
+                    <ChoiceInput
+                      id={candidate.id}
+                      name={contest.id}
+                      value={candidate.id}
+                      onChange={this.handleUpdateSelection}
+                      checked
+                      className="visually-hidden"
+                    />
+                    <Prose>
+                      <strong>{candidate.name}</strong>
+                    </Prose>
+                  </Choice>
+                )
+              })}
+            {!hasReachedMaxSelections && (
+              <Choice
+                as="button"
+                isSelected={false}
+                onClick={this.initWriteInCandidate}
+              >
+                <Prose>
+                  <em>add write-in candidate</em>
+                </Prose>
+              </Choice>
+            )}
           </Choices>
         </FieldSet>
         <Modal
-          isOpen={!!attemptedVoteCandidateName}
+          isOpen={!!attemptedOverVoteCandidate}
           content={
             <Prose>
-              <Text>
-                To vote for {attemptedVoteCandidateName}, first uncheck the vote
-                for {!!vote && vote.name}.
-              </Text>
+              {contest.seats === 1 ? (
+                <Text>
```

Comment:
nit: don't repeat what you don't need to repeat, e.g. `<Text>`.

Reply:
Simplified.